// Copyright 2018-2019 .
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.


syntax = "proto3";
package vsperf;

service Controller {
  rpc HostConnect (HostInfo) returns (StatusReply) {}
  rpc VsperfInstall (HostInfo) returns (StatusReply) {}
  rpc UploadConfigFile (stream ConfFile) returns (UploadStatus) {}
  rpc StartTest (ControlVsperf) returns (StatusReply) {}
  rpc TestStatus (StatusQuery) returns (StatusReply) {}
}

message ControlVsperf {
  string testtype = 1;
  string conffile = 2;
}

message ConfFile {
  bytes Content = 1;
}

message HostInfo {
  string ip = 1;
  string uname = 2;
  string pwd = 3;
}

message StatusQuery {
  string testtype = 1;
}

message StatusReply {
  string message = 1;
}

enum UploadStatusCode {
    Unknown = 0;
    Ok = 1;
    Failed = 2;
}

message UploadStatus {
    string Message = 1;
    UploadStatusCode Code = 2;
}
